File: ios/uclass/uclass/presentation/web/UclassJsInterface.swift
import Foundation
import WebKit

class UclassJsInterface: NSObject, WKScriptMessageHandler {
    private let onMessage: (String) -> Void
    
    init(onMessage: @escaping (String) -> Void) {
        self.onMessage = onMessage
        super.init()
    }
    
    // WKScriptMessageHandler 프로토콜 구현
    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
        if message.name == "uclass" {
            if let messageBody = message.body as? String {
                Logger.dev("웹에서 받은 메시지: \(messageBody)")
                onMessage(messageBody)
            }
        }
    }
    
    // JavaScript 코드를 반환하는 메서드 (웹에서 window.uclass.postMessage 사용 가능)
    static func getJavaScriptCode() -> String {
        return """
        window.uclass = {
            postMessage: function(message) {
                window.webkit.messageHandlers.uclass.postMessage(message);
            }
        };
        """
    }
}

File: android/UClass/app/src/main/java/com/ubase/uclass/presentation/web/UclassJsInterface.kt
package com.ubase.uclass.presentation.web

import android.content.Context
import android.webkit.JavascriptInterface

class UclassJsInterface(private val context: Context, private val onMessage:  (String) -> Unit) {

    @JavascriptInterface
    fun postMessage(message: String) {
        // 웹에서 호출되는 지점
        onMessage(message)
    }
}
